firestore/vector.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package firestore
+
+import (
+	"fmt"
+
+	pb "cloud.google.com/go/firestore/apiv1/firestorepb"
+)
+
+const (
+	typeKey       = "__type__"
+	typeValVector = "__vector__"
+	valueKey      = "value"
+)
+
+// Vector is an embedding vector.
+type Vector []float64
+
+// ToVector converts float32 or float64 slice to Firestore embedding vector.
+func ToVector[fType float32 | float64](arr []fType) Vector {
```

Comment:
Nit: Do we want add units tests for `ToVector()`in `vector_test.go`?

Reply:
Removed the method since it is longer required. Vector32 and Vector64 both datatypes will be supported by library